### PR TITLE
[Serve] Allow and remove trailing slashes in Ray submission address

### LINKED ISSUE #25969: [Serve] Serve CLI commands fail if address has a trailing slash
### What happened + What you expected to happen

The Serve CLI commands fail when specifying an address with a trailing slash (e.g. `http://localhost:8265/` fails when `http://localhost:8265` succeeds). The trailing slash should be dropped.

### Versions / Dependencies

Ray on the latest master.

### Reproduction script

In terminal, run:

1. `ray start --head`
2. `serve status`
```console
Traceback (most recent call last):
  File "/Users/shrekris/miniforge3/envs/ae/bin/serve", line 33, in <module>
    sys.exit(load_entry_point('ray', 'console_scripts', 'serve')())
  File "/Users/shrekris/miniforge3/envs/ae/lib/python3.8/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/shrekris/miniforge3/envs/ae/lib/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/shrekris/miniforge3/envs/ae/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/shrekris/miniforge3/envs/ae/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/shrekris/miniforge3/envs/ae/lib/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/shrekris/Desktop/ray/python/ray/serve/scripts.py", line 320, in status
    app_status = ServeSubmissionClient(address).get_status()
  File "/Users/shrekris/Desktop/ray/python/ray/dashboard/modules/serve/sdk.py", line 40, in __init__
    self._check_connection_and_version(
  File "/Users/shrekris/Desktop/ray/python/ray/dashboard/modules/dashboard_sdk.py", line 209, in _check_connection_and_version
    raise RuntimeError(version_error_message)
RuntimeError: Serve CLI is not supported on the Ray cluster. Please ensure the cluster is running Ray 1.12 or higher.
```

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change removes trailing slashes from a Ray submission address, rather than raising an `Exception`. This reduces confusion over the format required for a Ray submission address.

## Related issue number

Closes #25969.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - This change modifies some existing unit tests in `test_job_submission.py` and `test_cli.py` to submit to addresses with a trailing slash.

### CODE REVIEW

File: dashboard/modules/dashboard_sdk.py [hunk 1]
```diff
@@ -188,7 +188,14 @@ def __init__(
...
         )
-        self._address = cluster_info.address
+
+        # Remove any trailing slashes
+        self._address = cluster_info.address.rstrip("/")
+        logger.debug(
+            "The submission address cannot contain trailing slashes. Removing "
+            f'them from the requested submission address of "{address}".'
+        )
+
```
Comment (simon-mo) [inline dashboard/modules/dashboard_sdk.py:198]: guard this with `if cluster_info.address.endswith("/")`?

Comment (shrekris-anyscale) [reply dashboard/modules/dashboard_sdk.py]: Good catch– I added a guard.

**Review by simon-mo — APPROVED** [review]

**Comment by shrekris-anyscale** [comment]
@simon-mo Fyi, this change will also affect the job submission path, since I'm modifying the parent class. Do you think that'll be okay?

**Comment by simon-mo** [comment]
I'm not sure, probably? @architkulkarni thoughts?

**Review by architkulkarni — APPROVED** [review]
Seems fine to me, it's just increasing the allowed addresses, so it shouldn't cause anything to break.

**Review by wuisawesome — APPROVED** [review]

**Pull request merged**